### PR TITLE
[firebase_auth]: Correct configuration for iOS SMS auth

### PR DESCRIPTION
**Pull request by christianalfoni**
There are critical details missing in this README. The text is taken from the `firebase_messaging` package, which holds the details necessary for getting SMS authentication to work correctly on iOS. 

It took us about a month to figure this out, so I really hope we rescue other brave developers from going down the same path as we did 👍 😄 

Keep up the great work on Flutter and Firebase!

## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by christianalfoni** [comment]
Not quite sure why `analyze` breaks here... was just a doc update? Also did not check on irrelevant checkboxes? Hope that is okay, or let me know :)

File: packages/firebase_auth/README.md [hunk 1]
```diff
@@ -119,10 +119,11 @@ After a successful authentication, you will receive a `FirebaseUser` object. You
...
 
-1. [Enable App verification](https://firebase.google.com/docs/auth/ios/phone-auth#enable-app-verification)  
+2. Generate the certificates required by Apple for receiving push notifications following [this guide](https://firebase.google.com/docs/cloud-messaging/ios/certs) in the Firebase docs. You can skip the section titled "Create the Provisioning Profile".
 
-**Note:** App verification may use APNs, if using a simulator (where APNs does not work) or APNs is not setup on the
-device you are using you must set the `URL Schemes` to the `REVERSE_CLIENT_ID` from the GoogleServices-Info.plist file.
+3. In Xcode, select Runner in the Project Navigator. In the Capabilities Tab turn on Push Notifications and Background Modes, and enable Background fetch and Remote notifications under Background Modes.
+
+4. Also in the Project Navigator, choose the Info Tab. The `URL Types` section will need an entry where the `URL Schemes` has the value of your `REVERSE_CLIENT_ID` from the GoogleServices-Info.plist file. This ensures that ReCaptcha is triggered when  developing locally or device is not configured for silent push notifications
```
Comment (kroikie) [inline packages/firebase_auth/README.md]: nit: please use `1.` for numbered lists.

Comment (kroikie) [reply packages/firebase_auth/README.md]: nit: double space between `when` and `developing`

**Comment by christianalfoni** [comment]
Updated based on your feedback 😄 Sorry for late response, the email said `merged` for some reason. Weird.

**Comment by yukitaka13-1110** [comment]
@kroikie Why isn't this pull request merged?
I think this PR should be merged early.

**Comment by Ehesp** [comment]
Covered here https://firebase.flutter.dev/docs/auth/phone

**Pull request closed without merging**